sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -61,14 +61,13 @@ internal DiagnosticScope(string scopeName, DiagnosticListener source, object? di
                 return null;
             }
 
+            string clientName = ns;
             int indexOfDot = name.IndexOf(".", StringComparison.OrdinalIgnoreCase);
-            if (indexOfDot == -1)
+            if (indexOfDot != -1)
             {
-                return null;
+                clientName += "." + name.Substring(0, indexOfDot);
```

Comment:
Don't we still need to include the name in clientName even if it didn't have a dot?

Reply:
`clientName` becomes ActivitySource name, `name` will go to the Activity Name which will be created bu this source

Reply:
So, for the example from the ref docs:
ns: Azure.Storage.Blobs
name: BlobClient.DownloadTo
activity source name: Azure.Storage.Blobs.BlobClient
activity name: BlobClient.DownloadTo?

But if the name does not contain a dot:
ns: Azure.Messaging.ServiceBus
name: Message
activity source name: Azure.Messaging.ServiceBus
activity name: Message?

Is this what we want, e.g. not including the name in the activity source name only if there no dot? We repeat the first part of the name in both activity source name and activity name for the dot case, so this strikes me as somewhat inconsistent.

Reply:
> So, for the example from the ref docs:
ns: Azure.Storage.Blobs
name: BlobClient.DownloadTo
activity source name: Azure.Storage.Blobs.BlobClient
activity name: BlobClient.DownloadTo?

ns: namespace
name == ActivityName: class.method
activitySourceName: namespace.class // logic with dot does it.

For ServiceBus `Message` there is no class. So assuming class is optional, I don't see any inconsistency. I can totally miss something. Can you explain what you see?

Reply:
Interesting, I was thinking of Message as being the Class but I guess it is more like the operation (or method). Thanks for the explanation!